Lib/test/test_platform.py

```diff
@@ -229,6 +229,18 @@ def test_uname(self):
         self.assertEqual(res[-1], res.processor)
         self.assertEqual(len(res), 6)
 
+    @unittest.skipUnless(sys.platform.startswith('win'), "windows only test")
+    def test_uname_win32_without_wmi(self):
+        old_wmi_query = platform._wmi_query
+        def raises_oserror(*a):
+            raise OSError()
+        platform._wmi_query = raises_oserror
```

Comment:
You can use support.swap_attr() or unittest.mock.patch() for that, the context manager makes the scope of the change even more obvious (and it avoids to store explicitly the old value).